version: '3.1'

services:
  app:
    container_name: adressbook-app
    build:
      context: .
      dockerfile: Dockerfile
    ports:
      - 5000:5000
    volumes:
      - .:/app
    networks:
      - backend-network
    links:
      - mysqldb:mysql
    depends_on:
      mysqldb:
        condition: service_healthy

  mysqldb:
    container_name: adressbook-db
    image: mysql:8.0
    command: --default-authentication-plugin=mysql_native_password
    restart: always
    environment:
      MYSQL_DATABASE: dockerdb
      MYSQL_USER: adressbook
      MYSQL_PASSWORD: root
      MYSQL_ROOT_PASSWORD: root
    volumes:
      - ./dbdata:/var/lib/mysql
      - ./db:/docker-entrypoint-initdb.d/:ro
    networks:
      - backend-network
    ports:
      - 3307:3306
    healthcheck:
      test: ["CMD", "mysqladmin" ,"ping", "-h", "localhost"]
      timeout: 10s
      retries: 10        

volumes:
  dbdata:
    driver: local

networks:
  backend-network:
    driver: bridge